proposals/transcoding_invalid_identifiers/README.md

```diff
@@ -0,0 +1,316 @@
+  + Transform any UTF-8 string into a valid OpenUSD XID identifier. Mostly used for OpenUSD 24.03 and higher.
+
+* `std::string SdfBootstringDecodeIdentifier(const std::string&)`
+  * Transform the results of either `SdfBoostringEncodeAsciiIdentifier` or `SdfBoostringEncodeIdentifier` into the 
+  original UTF-8 string.
+
+# Proposed algorithm
+
+The algorithm is a generalization of Punycode, known as bootstring. There are references to how the algorithm 
+work in the [RFC-3492](https://datatracker.ietf.org/doc/html/rfc3492), the following document summarize it and 
+may offer a more friendly explanation of the concepts.
+
+## Encoding procedure
+
+### Separation of basic codes
+
+In general, in the Bootstring algorithm we need to differentiate between basic codes and extended codes:
+- **Basic codes** are known a priori and are the valid codes supported in our application domain. 
+- **Extended codes** are every other code which is not in the basic code set.
+
+For example, in OpenUSD versions prior to 24.03, the basic codes would be compromised of the characters `[A-Za-z0-9_]` 
+whereas from 24.03, the basic codes will make up to the Unicode XID specification.
+
+The first step in encoding is separating the basic codes from the extended codes. The basic codes will be copied 
+directly to the string, since they already belong to the domain of valid characters and will not cause any problem.
+If no extended code exists, then the algorithm finishes here.
+
+| Identifier        | Group    | Value         |
+|-------------------|----------|---------------|
+| `012-345-678/9.0` | Basic    | `01234567890` |
+|                   | Extended | `--/.`        |
+
+### Encoding of extended codes
+
+If there are extended codes, we start by appending a delimiter character. A delimiter character is a character
+which belongs to the set of basic codes and help to differentiate between basic and extended codes. 
+The original specification uses dash (`-`), however in our implementation we will use underscore (`_`).
+
+| Prefix | Basic codes   | Delimiter | Suffix              |
+|--------|---------------|-----------|---------------------|
+| `tn__` | `01234567890` | `_`       | `ENCODING_OF(--/.)` |
+
+
+We then use the following:
+
+#### Delta encoding
+
+Delta encoding is the process of encoding differences of values, instead of encoding directly the values. In transcoding
+this is useful as it exploits the fact that `UTF-8` characters of the same language appear close to each other.
+For example for japanese, Hiragana syllabary appears from `0x3040` to `0x309f` while Katakana appears from `0x30a0` to `0x30ff`.
+This helps to reduce the number of encoded bytes.
+
+| Characters sorted | Value to encode                            |
+|-------------------|--------------------------------------------|
+| `-`               | 45                                         |
+| `-`               | 0 (ASCII value 45, same as previous)       |
```

Comment:
Should this technically be encoding UTF-8 values?  (I know it's the same value.  Just talking about how they're labeled.)

Reply:
Yes, changed.